File: Data Structures & Algorithms/validate-parentheses/submission-2.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> S;
        for (char c : s) {
            if (c == '(' || c == '{' || c == '[') {
                S.push(c);
                continue;
            }
            if (c == ')' && (S.empty() || S.top() != '(')) return false;
            if (c == '}' && (S.empty() || S.top() != '{')) return false;
            if (c == ']' && (S.empty() || S.top() != '[')) return false;
            S.pop();
        }
        return S.empty();
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stack: Vec<i32> = Vec::new();
        for t in tokens.iter() {
            match t.as_str() {
                "+" => {
                    let a = stack.pop().unwrap();
                    let b = stack.pop().unwrap();
                    stack.push(b + a);
                }
                "-" => {
                    let a = stack.pop().unwrap();
                    let b = stack.pop().unwrap();
                    stack.push(b - a);
                }
                "*" => {
                    let a = stack.pop().unwrap();
                    let b = stack.pop().unwrap();
                    stack.push(b * a);
                }
                "/" => {
                    let a = stack.pop().unwrap();
                    let b = stack.pop().unwrap();
                    stack.push(b / a);
                }
                _ => stack.push(t.parse().expect(""))
            }
        }
        *stack.last().unwrap()
    }
}
